service/src/main/java/dk/dbc/search/solrdocstore/FrontendAPIBean.java

```diff
@@ -46,6 +46,22 @@ public Response getBibliographicKeys(@PathParam("bibliographicRecordId") String
         return Response.ok(new FrontendReturnListType<>(res),MediaType.APPLICATION_JSON).build();
     }
 
+    @GET
+    @Path("getBibliographicRecords/repositoryId/{repositoryId}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getBibliographicKeysByRepositoryId(@PathParam("repositoryId") String repositoryID) {
+
+        log.info("Requesting bibliographic record with repository id: {}",repositoryID);
+
+        // Escaping quotes to avoid parameter injections
+        String param = "{\"rec.repositoryId\":[\""+repositoryID.replaceAll("\"","\\\"")+"\"]}";
+        List<BibliographicEntity> res = entityManager.createNativeQuery("SELECT b.* FROM bibliographicsolrkeys b WHERE b.indexkeys @> ?::jsonb",BibliographicEntity.class)
+                .setParameter(1,param)
+                .setHint("javax.persistence.loadgraph",entityManager.getEntityGraph("bibPostWithIndexKeys"))
+                .getResultList();
```

Comment:
En eller anden form for paging med:
setFirstResult()
setMaxResults()

Reply:
Og nok også en ORDER BY ;-)

Reply:
enig